lib/_processQueue.ts

```diff
@@ -14,18 +16,29 @@ export function processQueue(globalObject) {
     const queue = globalObject[pointer].queue || [];
 
     // Loop queue and execute functions in the queue
-    queue.forEach((fn: string[]) => {
-      const functionName = fn[0];
-      const functionArguments = fn[1];
+    queue.forEach((fn: any[]) => {
+      const functionName: string = fn[0];
+      const functionArguments: any = fn[1];
+      const functionCallback: (err: any, res: any) => void = fn[2];
 
-      if (functionName && typeof (this as any)[functionName] === "function") {
-        this[functionName](functionArguments);
+      if (functionName && isFunction((this as any)[functionName])) {
+        const output: any = this[functionName](functionArguments);
+        if (isFunction(functionCallback)) {
+          functionCallback(null, output);
```

Comment:
Why don't we pass the error to the callback?

Reply:
There are currently some inconsistencies in a sense that some errors will not fire if comited before processQueue, but do after. 

Theses are inconsistencies I would like to tackle in later PR. They would probably create a big diff. I wanted to keep this PR about "how to get user Token" or more generally "how to get results from a method".